obspy/io/gcf/libgcf.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# reads Guralp Compressed Format (GCF) Files
+# By Ran Novitsky Nof @ BSL, 2016
+# ran.nof@gmail.com
+# Based on Guralp's GCF reference (GCF-RFC-GCFR, Issue C, 2011-01-05)
+# more details available from: http://www.guralp.com/apps/ok?doc=GCF_Intro
+# last access: June, 2016
+
+
+import numpy as np
+from obspy import UTCDateTime
+
+SPSD = {  # Table 3.1: special sample rates
+        157: 0.1,
+        161: 0.125,
+        162: 0.2,
+        164: 0.25,
+        167: 0.5,
+        171: 400,
+        174: 500,
+        176: 1000,
+        179: 2000,
+        181: 4000
+        }
+ToffsetsD = {  # Table 3.1: Time fractional offset denominator
+        171: 8.,
+        174: 2.,
+        176: 4.,
+        179: 8.,
+        181: 16.
+        }
+compressionD = {  # Table 3.2: format field to data type
+        1: '>i4',
+        2: '>i2',
+        4: '>i1'
+        }
+bitsD = {  # Table 3.2: format field to data bytes
+        1: 4,  # 32bits
+        2: 2,  # 16bits
+        4: 1   # 8bits
+        }
+
+
+def is_gcf(f):
+    'Test if file is GCF by reading at least 1 data block'
+    while 1:
+        header, data = read(f)
+        if len(data):
+            break
+
+
+def decode36(DATA):
+    'Decode base 36 data'
+    # http://geophysics.eas.gatech.edu/GTEQ/Scream4.4/Decoding_Base_36_numbers_C.htm
+    STR = ''
+    while DATA:
+        imed = DATA % 36
+        if imed > 9:
+            imed += 7
+        STR = chr(imed + 48) + STR
+        DATA = DATA / 36
+    return STR
+
+
+def decodeTime(DATA):
+    'Decode time field'
+    H = DATA / 3600
+    M = (DATA % 3600) / 60
+    S = DATA % 60
+    return "%02d:%02d:%02d" % (H, M, S)
```

Comment:
See example 4 of [the UTCDateTime constructor](https://docs.obspy.org/packages/autogen/obspy.core.utcdatetime.UTCDateTime.html).